src/components/UserManagement/UserTableData.jsx

```diff
@@ -9,24 +9,32 @@ import { faCopy } from '@fortawesome/free-solid-svg-icons';
 import { toast } from 'react-toastify';
 import { boxStyle } from 'styles';
 import { connect } from 'react-redux';
+// import { useHistory } from "react-router-dom";
 import { formatDate } from 'utils/formatDate';
 
 /**
  * The body row of the user table
  */
 const UserTableData = React.memo(props => {
   const [isChanging, onReset] = useState(false);
-  const canAddDeleteEditOwners = props.hasPermission('addDeleteEditOwners');
+  // const history = useHistory();
+  const canAddDeleteEditOwners = hasPermission('addDeleteEditOwners');
+  // const canAddDeleteEditOwners = props.hasPermission('addDeleteEditOwners');
 
+  console.log("user table", props)
   /**
    * reset the changing state upon rerender with new isActive status
    */
+
   useEffect(() => {
     onReset(false);
   }, [props.isActive, props.resetLoading]);
 
   const checkPermissionsOnOwner = () => {
-    return props.user.role === 'Owner' && !canAddDeleteEditOwners;
+    return (
+      props.user.role === 'Owner' &&
+      !hasPermission(props.role, 'addDeleteEditOwners', props.roles, props.userPermissions)
+    );
```

Comment:
```suggestion
    return props.user.role === 'Owner' && !canAddDeleteEditOwners;
```
`!hasPermission()` will always return false as `hasPermission()` returns a thunk. To get the desired boolean output, you need to call `props.hasPermission()` from `mapDispatchToProps`.

Reply:
Still need this change.